app/logic/adminNewEvent.py

```diff
@@ -4,6 +4,34 @@
 from app.models import mainDB
 from datetime import *
 from app.models.facilitator import Facilitator
+from app.controllers.admin import admin_bp
+from flask import request
+import json
+
+
+def calculateRecurringEventFrequency(recurringEventInfo):
+    """
+    """
+
+    eventName = recurringEventInfo['eventName']
+
+    endDate = datetime.strptime(recurringEventInfo['eventEndDate'], '%m-%d-%Y')
+    startDate = datetime.strptime(recurringEventInfo['eventStartDate'], '%m-%d-%Y')
+
+    recurringEvents = []
+
+    if endDate == startDate:
+        raise Exception("This event is not a recurring Event")
+
+    counter = 0
+    for i in range(0, ((endDate-startDate).days +1), 7):
+        counter += 1
+        recurringEvents.append({'eventName': f"{eventName} Week {counter}",
+                                'Date':startDate.strftime('%m-%d-%Y'),
+                                "week":counter})
+        startDate += timedelta(days=7)
+
+    return json.dumps(recurringEvents)

```

Comment:
I think this function should return recurringEvents straight up, and let the controller do the json conversion